cpu/lpc2387/periph/rtc.c

```diff
@@ -56,10 +53,12 @@ void rtc_init(void)
 
     RTC_CCR = CCR_CLKSRC;                   /* Clock from external 32 kHz Osc. */
 
-    /* initialize clock with valid unix compatible values
-     * If RTC_YEAR contains an value larger unix time_t we must reset. */
-    if (RTC_YEAR > 2037) {
-        _rtc_set(0);
+    /* Initialize clock with valid unix compatible values
+     * after cold boot or external reset.
```

Comment:
Nitpick: I would drop the whole "unix compatible", as this raises more questions than it answers. I'd rather just say that the clock is reset to a sane and predictable default after a cold boot or external reset.